File: apps/web/src/api/rules.ts
import request from '@/utils/request'
import type { Rule, RuleVersion, PaginatedResponse, PaginationParams } from '@/types'

export const getRuleList = (params?: PaginationParams): Promise<PaginatedResponse<Rule>> => {
  return request.get('/rules', { params })
}

export const getRuleDetail = (id: string): Promise<Rule> => {
  return request.get(`/rules/${id}`)
}

export const createRule = (data: Partial<Rule>): Promise<Rule> => {
  return request.post('/rules', data)
}

export const updateRule = (id: string, data: Partial<Rule>): Promise<Rule> => {
  return request.put(`/rules/${id}`, data)
}

export const deleteRule = (id: string): Promise<void> => {
  return request.delete(`/rules/${id}`)
}

export const updateRuleStatus = (id: string, status: number): Promise<Rule> => {
  return request.put(`/rules/${id}/status`, { status })
}

export const getRuleVersions = (id: string): Promise<RuleVersion[]> => {
  return request.get(`/rules/${id}/versions`)
}

export const createRuleVersion = (id: string, data: Partial<RuleVersion>): Promise<RuleVersion> => {
  return request.post(`/rules/${id}/versions`, data)
}

export const publishVersion = (ruleId: string, versionId: string): Promise<void> => {
  return request.post(`/rules/${ruleId}/versions/${versionId}/publish`)
}

export const rollbackVersion = (ruleId: string, versionId: string): Promise<void> => {
  return request.post(`/rules/${ruleId}/versions/${versionId}/rollback`)
}

export const cloneRule = (id: string): Promise<Rule> => {
  return request.post(`/rules/${id}/clone`)
}

export const exportRule = (id: string): Promise<Blob> => {
  return request.get(`/rules/${id}/export`, { responseType: 'blob' })
}

export const simulateRule = (data: Record<string, any>): Promise<Record<string, any>> => {
  return request.post('/rules/simulate', data)
}

File: apps/api/src/routes/rules.rs
use axum::{
    routing::{delete, get, post, put},
    Router,
};

use crate::AppState;
use crate::handlers::rules;
use crate::middleware::auth::require_auth;

pub fn router() -> Router<AppState> {
    Router::new()
        .route("/", get(rules::list).post(rules::create))
        .route("/:id", get(rules::detail).put(rules::update).delete(rules::delete))
        .route("/:id/status", put(rules::update_status))
        .route("/:id/versions", get(rules::list_versions).post(rules::create_version))
        .route("/:id/versions/:version_id/publish", post(rules::publish_version))
        .route("/:id/versions/:version_id/rollback", post(rules::rollback_version))
        .route("/:id/clone", post(rules::clone_rule))
        .route("/:id/export", get(rules::export_rule))
        .route("/export", get(rules::export_all_rules))
        .route("/simulate", post(rules::simulate))
        .layer(axum::middleware::from_fn(require_auth))
}
